Make build on Windows

guyer opened the pull request:
As was done for conda-forge/gmsh-feedstock

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

guyer committed:
Add fortran compilers

As was done for conda-forge/gmsh-feedstock

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer committed:
Ping the build

guyer committed:
Force use of openblas instead of mkl

Won't work because everything about conda and conda-forge is
a horrible nightmare from which it is impossible to awake.

guyer commented:
@conda-forge-admin, please update circle

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I just wanted to let you know that I updated the circle-ci deploy key and followed the project.

guyer committed:
Ping the build

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer committed:
Specify older version of openblas

guyer committed:
Fix package specification

I told you it wouldn't work

guyer committed:
Skip VC9

guyer commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.03

guyer committed:
Fiddle with logic of skipping different conditions

guyer committed:
Merge branch 'addfortran' of https://github.com/guyer/pysparse-feedstock into addfortran

guyer commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

guyer committed:
Remove VC9 skip

Bad idea, but trying to see what AppVeyor isn't building at all

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.03

guyer committed:
Revert "Remove VC9 skip"

This reverts commit 7ff0cbc165ae9a0ed88c6ecdd4105ed6d5844139.

guyer committed:
Merge branch 'addfortran' of https://github.com/guyer/pysparse-feedstock into addfortran

guyer commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

guyer commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.08

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer committed:
Remove Fortran and VC14 requirements

guyer commented:
@conda-forge-admin, please rerender

conda-forge-admin committed:
MNT: Re-rendered with conda-smithy 3.1.12 and pinning 2018.10.08

isuruf committed:
Update meta.yaml

guyer commented:
@isuruf Thanks for your efforts. It's [now failing](https://ci.appveyor.com/project/conda-forge/pysparse-feedstock/builds/19419276) because it doesn't find blas and lapack, as near as I can tell, even though it installs `mkl`.

When this was a [staged-recipe](https://github.com/conda-forge/staged-recipes/pull/6586), it [passed](https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.27812), finding `mkl_rt` for blas and lapack.

Any idea why it worked then, but not now?

guyer committed:
Force use of older mkl

blas and lapack [don't get found](https://ci.appveyor.com/project/conda-forge/pysparse-feedstock/builds/19419276)
with mkl 2019.0

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [29]

guyer committed:
Fix linting

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

guyer commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -27,10 +26,13 @@ requirements:
     - pip
     - distribute
+    - openblas        0.2.20   # [unix]
+    - mkl             2018.0.3  # [win]
```
Please read [this doc about `blas`]( https://conda-forge.org/docs/blas.html ).

Also pinnings are handled at a global level. So no need to pin here.

We currently haven't figured out MKL support due to legal gray areas, but are looking into it. Though there is `openblas` on Windows.

guyer replied:
@jakirkham I've read [the `blas` link you sent](https://conda-forge.org/docs/blas.html). Was it sent on a google translate round-robin, by any chance? 

The situation with this feedstock is this:
- PySparse is compatible with Python 2.x only
- Python 2.7 is apparently only available with `vc9` on `[win]`
- `openblas` is apparently only available with `vc14` on `[win]`
- I pinned `openblas` on `[unix]` at 0.2.20 because my feedstock ceased to build when `openblas` 0.3.3 went up
- as a result, on the recommendation of @isuruf, I required `mkl` on `[win]`
- PySparse' `setup.py` [couldn't find `mkl`](https://ci.appveyor.com/project/conda-forge/pysparse-feedstock/builds/19419276), even though it could [when this was a `staged-recipe`](https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.27812)
- as a result, I'm trying to pin `mkl` to 2018.0.3, which is what worked ast a `staged-recipe`
- AppVeyor sits on every one of my changes for hours, making troubleshooting this virtually impossible
- I've spent months banging my head against these feedstocks and I really hate conda-forge right now

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.
